server/pulp/server/controllers/content.py

```diff
@@ -176,14 +177,30 @@ def _download_catalog_entry(catalog_entry):
     if downloader.event_listener.succeeded_reports:
         _logger.info(_('Download {url} via {streamer_url} succeeded').format(
             url=catalog_entry.url, streamer_url=request.url))
-        content_unit.set_content(request.destination)
+        content_unit.set_content(request.destination, downloaded=False)
```

Comment:
Yes, it's quite confusing. Suppose we're dealing with a multi-file content unit: we download one file and copy it to its final resting place, but we don't know if that's the last file for that content unit, so we have to say that it's false. Later, we check to see if all the files are present, and if they are, we set it to True.

This is clearly not what you were thinking about though, so maybe we should sync up.